CIP-XXXX/README.md

```diff
@@ -0,0 +1,84 @@
+
+To solve this, one way would be to instead have a proxy contract that receives funds and forwards them to your DAO with the proper structure. Native scripts at the moment can play this role by creating a native script multisig where some set of DAO members have their public keys specified in the spending condition of the multisig. However, this approach has the following problems:
+
+- There is no way to guarantee that these members will actual forward the funds to the DAO instead of pocketing the funds
+- DAO membership may not easily be representable by a small set of fixed public keys
+
+These problems could be solved by adding a many new native script conditions such as enforcing that the transaction that spends it needs to have a certain NFT as part of its inputs or that the transaction contains a specific output. However, there is no guarantee this is flexible enough for all use-cases and these may bring unnecessary feature creep to the native script feature.
+
+Instead, the most generic solution is to allow a new condition where a native scripts can only be spent if a specific Plutus script is also part of the transaction input. This allows the multisig to simply handle receiving the funds and having all the complex logic (DAO membership checks, output checks, etc.) to be added to the Plutus script.
+
+# Specification
+
+The current definition of native scripts uses the following BNF notation
+
+```BNF
+<native_script> ::=
+             <RequireSignature>  <vkeyhash>
+           | <RequireTimeBefore> <slotno>
+           | <RequireTimeAfter>  <slotno>
+
+           | <RequireAllOf>      <native_script>*
+           | <RequireAnyOf>      <native_script>*
+           | <RequireMOf>        <num> <native_script>*
+```
+
+Importantly, note that native scripts can only require other native scripts and not plutus scripts or any other future script type introduced.
+
+Therefore, this proposal suggests the definition be changed to
+
+```BNF
+<native_script> ::=
+             <RequireSignature>  <vkeyhash>
+           | <RequireScript>     <scripthash>
+           | <RequireTimeBefore> <slotno>
+           | <RequireTimeAfter>  <slotno>
+
+           | <RequireAllOf>      <native_script>*
+           | <RequireAnyOf>      <native_script>*
+           | <RequireMOf>        <num> <native_script>*
+```
+
+Which we propose uses the `type: script` when used in JSON notation such as in the following example:
+
+```json
+{
+  "type": "all",
+  "scripts":
+  [
+    {
+      "type": "script",
+      "scriptHash": "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538"
+    },
+  ]
+}
+```
+
```

Comment:
It wouldn't hurt to add a section on how exactly the referred scripts are going to be interpreted. Here's what I would add.

```suggestion

## Script Execution

The proposed spending condition of `"type": "script"` may refer to any simple or Plutus script via the `scriptHash` field. The actual script is presented by the spending transaction: either in the witness set or in a reference input.

Depending on the type of the referred script, additional processing is done. 
- native (a.k.a. simple) scripts are recursively evaluated as described in https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/simple-scripts.md, with the addition of the `RequireScript` constructor proposed here.
- Plutus scripts are executed as regular spending script as (informally) described in https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/plutus/plutus-spending-script-example.md. This means, the output's datum is passed to the script, as well as the redeemer presented by the spending transaction, and script context (constructed according to the Plutus version) is passed as the third argument.
```

Reply:
Perhaps, it is bad wording. Would it be clearer if we remove the phrase `additional processing is done.` and keep the rest?